### PR TITLE
fix: fix bucket deletion issue when parameter changed

### PR DESCRIPTION
**Pull request by forcodedancing**
### Description

Fix bucket cannot be deleted issue when there are discontinued objects in it, it will happen when changing the deletion confirm period parameter smaller.

### Rationale

Bug fix 

### Example

NA

### Changes

Notable changes: 
* storage keeper

### CODE REVIEW

**Review by guagualvcha — APPROVED** [review]

File: x/storage/keeper/keeper.go [hunk 1]
```diff
@@ -1508,6 +1525,7 @@ func (k Keeper) getDiscontinueObjectStatus(ctx sdk.Context, objectId types.Uint)
 	store := ctx.KVStore(k.storeKey)
 	bz := store.Get(types.GetDiscontinueObjectStatusKey(objectId))
 	if bz == nil {
+		ctx.Logger().Error("fail to get discontinued object status", "object id", objectId)
```
Comment (Fynnss) [inline x/storage/keeper/keeper.go:1528]: Having delete logic(#1532 `store.Delete(types.GetDiscontinueObjectStatusKey(objectId))` ) inside the GetXXX function is not considered a good practice.

File: x/storage/keeper/keeper.go [hunk 2]
```diff
@@ -230,31 +231,46 @@ func (k Keeper) ForceDeleteBucket(ctx sdk.Context, bucketId sdkmath.Uint, cap ui
...
+		// which means that after deleting objects when deleting a bucket the objects in it should be in
+		// OBJECT_STATUS_CREATED or OBJECT_STATUS_SEALED status.
+		// However, when updating the deletion confirm period parameter, it can lead to that
+		// the discontinued bucket can be deleted earlier, then its objects could be in
+		// OBJECT_STATUS_DISCONTINUED status.
+
+		objectStatus := objectInfo.ObjectStatus
+		if objectStatus == types.OBJECT_STATUS_DISCONTINUED {
+			objectStatus, err = k.getDiscontinueObjectStatus(ctx, objectInfo.Id)
+			if err != nil {
```
Comment (Fynnss) [inline x/storage/keeper/keeper.go:244]: If an error occurs, does the object need to wait for the `discontinueObject` to be deleted?

File: x/storage/keeper/keeper.go (hunk 1 above)
Comment (forcodedancing) [reply x/storage/keeper/keeper.go:1528]: renamed the function to make it clear.

**Review by Fynnss — APPROVED** [review]
LGTM

**Pull request merged**